### PR TITLE
[SPRAL] add recipe

### PR DESCRIPTION
**Pull request by ViralBShah**
Try to build the SPRAL library.

@ueliwechsler

### CODE REVIEW

File: S/SPRAL/build_tarballs.jl [hunk 1]
```diff
@@ -68,18 +69,19 @@ platforms = supported_platforms()
 # The products that we will ensure are always built
 products = [
```
Comment (giordano) [inline S/SPRAL/build_tarballs.jl:70]: This argument must be `Vector{Product}`
```suggestion
products = Product[
```

**Comment by ViralBShah** [comment]
@ueliwechsler You can see the errors it has building on windows, which seem to be the same as other platforms. Perhaps should be easy to get fixed upstream if you can file the relevant issues there.

**Comment by ViralBShah** [comment]
Summary - there are issues with all 32-bit builds in the building of the METIS wrappers.
The mac and freebsd builds have a problem with finding `struct timespec` which presumably is a missing header file somewhere.

**Comment by ViralBShah** [comment]
I think the issues should be reported upstream and it appears they may not be too difficult for the devs to fix. If they are up for it, it would also be nice for them to tag a release, since we are building from master.

**Pull request closed without merging**